vm/ubpf_jit_x86_64.c

```diff
@@ -93,6 +93,11 @@ translate(struct ubpf_vm *vm, struct jit_state *state, char **errmsg)
         emit_mov(state, RDI, map_register(1));
     }
 
+    /* Move rsi into register 2 */
+    if (map_register(2) != RSI) {
+        emit_mov(state, RSI, map_register(2));
+    }
```

Comment:
Given that we can't rely on the length register when applying a register offset, I think these lines should just be removed.... Keeping them here just confusingly makes it seem like that would work.

Reply:
@jpsamaroo can you comment on this. It looks like @lsgunth is correct here.

Reply:
I just realized that we'll already be passing the memory length in RSI anyway thanks to calling conventions, so there's no need for this :smile:

Reply:
Thanks @jpsamaroo. I will drop these lines and resubmit.